---
title: "Processing fastq files to generate sequencing read count matrix"
author: "Evangelyn Sim"
date: "2/15/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```




## Introduction

Following sequencing and obtaining .fastq.gz file, the first step is to perform trimming and mapping of the sequencing data to generate bam files. All these steps were performed using bash code. 

Bam files were then used for read counts to generate a count matrix.

Mouse AAV-PGR bulk RNA-seq were performed using paiered end sequencing method and below are the scripts for trimming and mapping paired end sequencing read.


### Used libraries and functions

* skewer/0.2.2
* star/2.5.3a
* samtools/1.8
* parallel
* subread/1.5.0


### Trimming of sequencing read

    #!/bin/bash

    # function to run skewer quality trimming
    runskew(){
    FQZ1=$1
    FQZ2=`echo $FQZ1 | sed 's/_R1.fastq.gz/_R2.fastq.gz/'`
    skewer -t 8 -q 20 $FQZ1 $FQZ2
    }
    export -f runskew

    # actually run skewer
    parallel -j3 runskew ::: *_R1.fastq.gz


### Mapping of Skewer trimmed .fastq to mouse reference genome

It will generate the following 4 outputs for individual .fastq.gz file:

1. .STAR.bam
2. .STAR.bam.bai
3. .STAR.bam.stats
4. _starlog.txt

```{bash}

#!/bin/bash

DIR=/group/card2/Evangelyn_Sim/Sequencing_ATAC_RNA/refgenome/star
GTF=/group/card2/Evangelyn_Sim/Sequencing_ATAC_RNA/refgenome/Mus_musculus.GRCm38.96.gtf

for FQ1 in `ls *1.fastq-trimmed-pair1.fastq` ; do
    FQ2=`echo $FQ1 | sed 's/1.fastq-trimmed-pair1.fastq/1.fastq-trimmed-pair2.fastq/'`
    BASE=`echo $FQ1 | sed 's/_1.fastq-trimmed-pair1.fastq//'`

STAR --genomeLoad NoSharedMemory --genomeDir $DIR --readFilesIn $FQ1 $FQ2 --runThreadN 30 \
--sjdbGTFfile $GTF --outSAMattributes NH HI NM MD

rm $FQ1 $FQ2
mv Aligned.out.sam ${BASE}.STAR.sam
mv Log.final.out ${BASE}_starlog.txt

( samtools view -uSh ${BASE}.STAR.sam | samtools sort -o ${BASE}.STAR.bam
rm ${BASE}.STAR.sam
samtools index ${BASE}.STAR.bam
samtools flagstat ${BASE}.STAR.bam > ${BASE}.STAR.bam.stats ) &

done

STAR genomeLoad Remove --genomeDir $DIR
wait


```

### Merge bam files

Make a directory called "merged" and ln all .bam files to the folder and perform the following.

```{bash}
#!/bin/bash

samtools view -H `ls *bam | head -1` > header.sam
for BASE in `ls *bam | cut -d '_' -f2 | sort -u ` ; do
  rm $BASE.mg.bam
  samtools merge -h header.sam $BASE.mg.bam *${BASE}*bam &
done
wait


```


### Counting reads from bam files across mouse reference genome

```{bash}

#!/bin/bash

SAF=/group/card2/Evangelyn_Sim/Sequencing_ATAC_RNA/refgenome/Mus_musculus.GRCm38.96.fulllength.saf
OUT=mrna_fulllen_pe_strrev_q30.mx

#featureCounts -p -Q 30 -T 20 -s 2 -a $SAF -F SAF -o $OUT *bam


```



### Tidy counted matrix

```{bash}

#!/bin/bash

for MX in `ls *mx` ; do
   sed 1d $MX | sed 's/.mg.bam//g' > $MX.all
   sed 1d $MX | cut -f1-6 | sed 's/.mg.bam//g' > $MX.chr
   sed 1d $MX | cut -f1,7- | sed 's/.mg.bam//g' > $MX.PR.fix


done
wait

```

### Filter out low counts genes from matrix

Filtering out low counts genes by running the following filter.sh as 

  bash filter.sh hrna_dev_mf_fulllen_se_strrev_q30.mx.all.fix

  filter.sh

    head -1 $1 > ${1}_filt
    awk '{
      min = max = sum = $2;       # Initialize to the first value (2nd field)
      sum2 = $2 * $2              # Running sum of squares
      for (n=3; n <= NF; n++) {   # Process each value on the line
        if ($n < min) min = $n    # Current minimum
        if ($n > max) max = $n    # Current maximum
        sum += $n;                # Running sum of values
        sum2 += $n * $n           # Running sum of squares
      }
      print sum/(NF-1) ;
    }' $1 > avg
    paste avg $1 | awk '$1 >= 10' | cut -f2- | tr ' ' '\t' >> ${1}_filt
    rm avg
