---
title: "MDS plot"
author: "Evangelyn Sim"
date: "2/15/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```




## Introduction

In the GEO submission 2 processed files were uploaded. 

1. mrna_fulllen_pe_strrev_q30.mx.PRKO.unfiltered
2. mrna_fulllen_pe_strrev_q30.mx.PRKO.all.fix_filt

They have been uploaded in the /output folder and will be used below to generate different figures.



### Used libraries and functions

```{r}
library(edgeR)
library(limma)
library(Glimma)
library(gplots)

library(ComplexHeatmap)
library(circlize)
library(RColorBrewer)
library(mclust)

```


### Read files 
```{r}

rm1 <- read.delim("/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/GITHUB/Mouse_PRKO_RNAseq_bulk/output/mrna_fulllen_pe_strrev_q30.mx.PRKO.all.fix_filt", row.names = 1)

info = read.delim("/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/GITHUB/Mouse_PRKO_RNAseq_bulk/output/20200211_PRKO_RNA_samplesheet.txt", header = TRUE, sep = "\t")
info$ID = gsub("-",".",info$ID)

names(rm1) = info$ID[match(names(rm1),info$ID)]

m = match(info$ID,names(rm1))
rm2 = rm1[,m]
rm1 = rm2


sampleinfo = info
levels(factor(sampleinfo$Group))
levels(factor(sampleinfo$BinSex))
sampleinfo$colour = c("aquamarine2","mediumpurple1")[factor(sampleinfo$Group)]
table(colnames(rm2)==sampleinfo$ID)

y <- DGEList(rm2)



```

# Make plot

```{r, fig.height=4,fig.width=10}

par(mfrow=c(1,3))

plotMDS(y, pch=c(0,1)[factor(sampleinfo$Group)], col=sampleinfo$colour,dim.plot = c(1,2), cex = 2)
legend("top", legend = c("AAV_MCS","AAV_PR"), pch=c(1,0), col = c("mediumpurple1", "aquamarine2"), cex=1)

plotMDS(y, pch=c(0,15,1,16)[factor(sampleinfo$BinSex)], col=sampleinfo$colour, dim.plot = c(1,2),cex = 2)
legend("top", legend = c("AAV_MCS","AAV_PR"), pch=c(1,0), col = c("mediumpurple1", "aquamarine2"), cex=1)
legend("topleft", legend = c("F","M"), pch=c(1,16), col = c("grey"), cex=1)

plotMDS(y, cex = 0.8, dim.plot = c(1,2))



```


# Remove Chr X & Y genes

```{r}

rm3 = read.delim("/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/GITHUB/Mouse_PRKO_RNAseq_bulk/output/mrna_fulllen_pe_strrev_q30.mx.chr", header = TRUE, row.names = 1)
rm3 = merge(rm3, rm2, by = "row.names")
rm3 = rm3[!grepl("Y",rm3$Chr),]
rm3 = rm3[!grepl("X",rm3$Chr),]
row.names(rm3)= rm3$Row.names
rm3 = rm3[,c(7:ncol(rm3))]
rm2 = rm3

z <- DGEList(rm3)

```

# Make MDS plot (removed Chr X & Y genes)

```{r, fig.height=4,fig.width=10}

par(mfrow=c(1,3))

plotMDS(z, pch=c(0,1)[factor(sampleinfo$Group)], col=sampleinfo$colour,dim.plot = c(3,4), cex = 2)
legend("topright", legend = c("AAV_MCS","AAV_PR"), pch=c(1,0), col = c("purple", "springgreen3"), cex=1)

plotMDS(z, pch=c(0,15,1,16)[factor(sampleinfo$BinSex)], col=sampleinfo$colour, dim.plot = c(3,4),cex = 2)
legend("topright", legend = c("AAV_MCS","AAV_PR"), pch=c(1,0), col = c("purple", "springgreen3"), cex=1)
legend("bottomright", legend = c("F","M"), pch=c(1,16), col = c("grey"), cex=1)

plotMDS(z, cex = 0.8, dim.plot = c(3,4))


```
